core/spec/lib/spree/preferences/preference_differentiator_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'spree/preferences/preference_differentiator'
+require 'spree/preferences/configuration'
+
+RSpec.describe Spree::Preferences::PreferenceDifferentiator do
+  it 'includes defaults that have changed' do
```

Comment:
Should we add a test for new options too?

Reply:
This class takes a single `Configuration` class, which can define values that have changed over time. They are defined with the `by_version` method, which takes an initial value and boundaries where it changed. E.g.:

```ruby
by_version(true, "3.0.0" => false)
```

If we introduce a new preference on Solidus that breaks some behavior, it'll be equivalent to say that the preference had a different value on past versions, so the `by_version` definition could be the same as the previous example.